libgalois/include/galois/worklists/AdaptiveObim.h

```diff
@@ -292,56 +349,50 @@ struct AdaptiveOrderedByIntegerMetric : private boost::noncopyable {
-      unsigned long numPushesThisStep      = 0;
-      unsigned long allPmodDeqCounts       = 0;
-      Index minOfMin                       = std::numeric_limits<Index>::max();
-      Index maxOfMax                       = std::numeric_limits<Index>::min();
-      for (unsigned i = 0; i < runtime::activeThreads; ++i) {
-        Index& otherMinPrio = current.getRemote(i)->minPrio;
-        minOfMin = compare(minOfMin, otherMinPrio) ? minOfMin : otherMinPrio;
-        Index& otherMaxPrio = current.getRemote(i)->maxPrio;
-        maxOfMax = compare(otherMaxPrio, maxOfMax) ? maxOfMax : otherMaxPrio;
-        priosCreatedThisPeriod += current.getRemote(i)->priosLastPeriod;
-        numPushesThisStep += current.getRemote(i)->pushesLastPeriod;
-        allPmodDeqCounts += current.getRemote(i)->pmodAllDeq;
-        current.getRemote(i)->sinceLastFix          = 0;
-        current.getRemote(i)->slowPopsLastPeriod    = 0;
-        current.getRemote(i)->pushesLastPeriod      = 0;
-        current.getRemote(i)->priosLastPeriod       = 0;
-        current.getRemote(i)->maxPrioDiffLastPeriod = 0;
-
-        current.getRemote(i)->minPrio = std::numeric_limits<Index>::max();
-        current.getRemote(i)->maxPrio = std::numeric_limits<Index>::min();
-      }
 
-      if (((double)numPushesThisStep /
-           ((double)((maxOfMax >> delta) - (minOfMin >> delta)))) <
-          chunk_size / 2) {
-        double xx = ((double)(chunk_size) /
-                     ((double)numPushesThisStep /
-                      ((double)((maxOfMax >> delta) - (minOfMin >> delta)))));
-        delta += std::floor(std::log2(xx));
-        counter *= 2;
+        Index& otherMinPrio = data.getRemote(i)->minPrio;
+        minOfMin            = std::min(minOfMin, otherMinPrio, compare);
+        Index& otherMaxPrio = data.getRemote(i)->maxPrio;
+        maxOfMax            = std::max(otherMaxPrio, maxOfMax, compare);
+        numPushesThisStep += data.getRemote(i)->pushesLastPeriod;
+        priosCreatedThisPeriod += data.getRemote(i)->stats.priosLastPeriod;
+        allPmodDeqCounts += data.getRemote(i)->stats.pmodAllDeq;
+
+        data.getRemote(i)->cleanup();
+        data.getRemote(i)->lock.unlock();
       }
 
-      for (unsigned i = 1; i < runtime::activeThreads; ++i) {
-        current.getRemote(i)->lock.unlock();
+      if ((double)numPushesThisStep) {
+        Index prioRange = (maxOfMax >> delta) - (minOfMin >> delta);
+        // Division is expensive
+        // double fillRatio = ((double)numPushesThisStep / (double)prioRange);
+        if (numPushesThisStep < (chunk_size >> 1) * prioRange) {
+          // Ditto
+          // double xx = ((double)(chunk_size) / fillRatio);
+          double xx = std::log2(chunk_size) - std::log2(numPushesThisStep) +
+                      std::log2(prioRange);
+          assert(xx);
+          delta += std::floor(xx);
+          counter <<= 1;
+        }
       }
     }
-#ifdef UNMERGE_ENABLED
```

Comment:
This block was originally disabled. Did you mean to re-enable it? I assume they disabled it for performance reasons.

Reply:
I made this macro-control a template parameter, which is still false/disabled by default. To me, it doesn't make sense to keep a statically disabled code block in the codebase.